blog/2018/12/12/dancer-and-minion/index.markdown

```diff
@@ -0,0 +1,299 @@
+    }); 
+}
+```
+
+## Creating and Configuring the Job Queue Worker
+
+We wanted to easily configure our Minions for all hosts and environments in one spot. Since we use a lot of YAML in Dancer, specifying the Minion configuration in YAML made a lot of sense
+to us:
+
+```
+# What port does the dashboard listen on?
+dashboard_port: 4000
+
+# Add the rest later.
+dashboards:
+    UNKNOWN: http://localhost:3000/
+    DEV: http://my.development.host.tld:8001/
+
+# Hosts that have no entry assume the default configuration
+default:
+    max_children: 4
+    queues:
+        - default
+
+# Host-specific settings
+jcrome-precision-3510:
+    max_children: 8
+    queues:
+        - default
+        - InstantXML
+        - PayrollXML
+        - ChangeRequest
+```
+Our job queue workers look like:
+```
+#!/usr/bin/env perl
+
+use MyJob::Base;
+use MyJob::JobQueue;
+use MyJob::Log4p;
+use MyJob::Util::Logger;
+use MyJob::Util::SysTools qw(get_hostname);
+
+use Mojo::mysql;
+
+my $config     = MyJob::Config->new->config;
+my $hostconfig = get_hostconfig();
+my $minion     = MyJob::JobQueue->new;
+my $worker     = $minion->runner->worker;
+
+my $log_eng = MyJob::Log4p->new({ logger_name => "Minion" });
+my $logger  = MyJob::Util::Logger->new->logger($log_eng);
+
+$worker->on( dequeue => sub( $worker, $job ) {
+    my $id    = $job->id;
+    my $notes = $job->info->{ notes };
+    my $title = $notes->{ title };
+    my $guid  = $notes->{ guid };
+
+    $job->on( spawn => sub( $job, $pid ) {  
+        $0 = "$title $guid";
+        $logger->info( "$title: Created child process $pid for job $id by parent $$ - $guid");
+    });
+    
+    $job->on( failed => sub( $job, $error ) {
+        chomp $error;
+        $logger->error( $error );
+    });
+});
+
+$worker->on( busy => sub( $worker ) {
+    my $max = $worker->status->{ jobs };
+    $logger->log( "$0: Running at capacity (performing $max jobs)." );
+});
+
+my $max_jobs = $hostconfig->{ max_children };
+my @queues   = @{ $hostconfig->{ queues }};
+
+if( $minion->has_invalid_queues( @queues ) ){
+    print "Invalid job queues specified: " . join( ',', $minion->get_invalid_queues( @queues ) );
+    say ". Aborting!";
+    exit 1;
+}
+
+say "Starting Job Queue Worker on " . get_hostname();
+say "- Configured to run a max of $max_jobs jobs";
+say "- Listening for jobs on queues: ", join(', ', @queues );
+$worker->status->{ jobs }   = $max_jobs;
+$worker->status->{ queues } = \@queues;
+$worker->run;
+
+sub get_hostconfig {
+    my $minion_config = MyJob::Config->new({ filename => "environments/minions.yml" })->config;
+    my $hostname      = get_hostname();
+
+    if( exists $minion_config->{ $hostname }) {
+        return $minion_config->{ $hostname };
+    } else {
+        return $minion_config->{ default };
+    }
+}
+```
+## Monitoring the Workers
+
+We made one of the simplest Mojolicious apps possible to run the Minion dashboard for us, which we then link to from our own dashboard:
```

Comment:
Link to the previous few advent articles as @preaction covers both having a tiny mojo script just to have minion and how to run the ui. Just quick links, you don't even need any text if you don't want. It just helps round out the "series" on Minion that has been going.

Reply:
Linked to @preaction's article.